pkg/collectorinterface/tokenauth/send.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright © 2021 The MayaData Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tokenauth
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mayadata-io/volume-events-exporter/pkg/collectorinterface"
+	"github.com/mayadata-io/volume-events-exporter/pkg/env"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// postMethod is used to send http POST request
+	postMethod   = "POST"
+	okStatusCode = 200
+)
+
+type TokenClient struct {
+	// server url holds the URL to communicate with server
+	serverURL string
+
+	// serverAuthToken holds the token of the server
+	serverAuthToken string
+
+	// dataType represents the type of the data that server
+	// can understand. Defaults to JSON
+	dataType collectorinterface.DataType
+	// Client to interact with server
+	client *http.Client
+	// VolumeCollector implements methods required for event collector
+	collectorinterface.VolumeEventCollector
+}
+
+func NewTokenClient(collectorInterface collectorinterface.VolumeEventCollector) collectorinterface.EventsSender {
+	return &TokenClient{
+		serverURL:            env.GetCallBackServerURL(),
+		serverAuthToken:      env.GetCallBackServerAuthToken(),
+		dataType:             collectorinterface.JSONDataType,
```

Comment:
> we can fetch it through collectorInterface since backend PV engine will be sending data in serialized json/yaml format

Do you mean by adding a different function in `collectorInterface`? Something like caller will look like `obj.Type(json/yaml).CollectCreateEvents()`?

Reply:
Updated as per discussion